### DISCUSSION TITLE
page duplication protocol for page migration

### DISCUSSION THREAD

**Discussion by Soilorian** [sarchlab/mgpusim#65] [Q&A] [answered]
I'm reaching out regarding some adjustments I'd like to make to the page migration policies in mGPU-Sim. I've successfully located the implementation for on-touch migration and am confident in my ability to implement counter-based migration based on that. However, I've encountered a challenge with page duplication. Specifically, I'm unsure how to handle writing to these duplicated pages. During a write operation, it's crucial to locate and update all copies of the page. I'm curious if there's an existing solution for this task or if I should proceed with implementing one myself. If you have any insights or guidance on how to efficiently identify all pages with the same address in storage, I would greatly appreciate it.

**Comment by syifan** [comment, marked as answer]
There is no built-in mechanism for writing into multiple locations at the same time. So my suggestion is that you need to think about how real hardware performs such actions.
